@utahdts/utah-design-system/react/components/forms/CalendarInput/CalendarInput.jsx

```diff
@@ -0,0 +1,389 @@
+import {
+  add,
+  format,
+  isValid,
+  parse
+} from 'date-fns';
+import {
+  useCallback,
+  useEffect,
+  useId,
+  useMemo,
+  useRef,
+  useState
+} from 'react';
+import { Button, IconButton, useAriaMessaging } from '../../../..';
+import { joinClassNames } from '../../../util/joinClassNames';
+import { useOnKeyUp } from '../../../util/useOnKeyUp';
+import { ErrorMessage } from '../ErrorMessage';
+import { useFormContextInputValue } from '../FormContext/useFormContextInputValue';
+import { RequiredStar } from '../RequiredStar';
+import { calendarGrid } from './calendarGrid';
+
+/** @typedef {import('@utahdts/utah-design-system').CalendarGridValue} CalendarGridValue */
+
+/**
+ * @template FormEventT
+ * @typedef {import('react').FormEvent<FormEventT>} FormEvent
+ */
+
+/**
+ * @param {string} calendarInputId
+ * @param {Date | null} newDate
+ * @param {string} dateFormat
+ */
+function moveCurrentValueFocus(calendarInputId, newDate, dateFormat) {
+  // focus on the next date; delay so that the new month view draws before it focuses
+  setTimeout(
+    () => {
+      if (newDate) {
+        const formattedDate = format(newDate, dateFormat);
+        document.getElementById(`${calendarInputId}__${formattedDate}`)?.focus();
+      }
+    },
+    0
+  );
+}
+
+/**
+ * @param {object} props
+ * @param {string} [props.className]
+ * @param {string} [props.dateFormat] use `date-fns` modifiers for formatting the date
+ * @param {string} [props.defaultValue] expects value to be in format of props.dateFormat
+ * @param {string} [props.errorMessage]
+ * @param {string} props.id when tied to a Form the `id` is also the 'dot' path to the data in the form's state: ie person.contact.address.line1
+ * @param {import('react').RefObject<HTMLDivElement>} [props.innerRef]
+ * @param {boolean} [props.isDisabled]
+ * @param {boolean} [props.isHidden] a dateInput will hide its calendar popup when not in use
+ * @param {boolean} [props.isRequired]
+ * @param {string} props.label
+ * @param {string} [props.labelClassName]
+ * @param {(newValue: string) => void} [props.onChange] e => {}; can be omitted for uncontrolled OR using form's onChange
+ * @param {boolean} [props.shouldSetFocusOnMount] if rendered in a popup, then set focus to first focusable element when first shown
+ * @param {boolean} [props.showTodayButton]
+ * @param {string | null} [props.value] expects value to be in format of props.dateFormat
+ * @param {string} [props.wrapperClassName]
+ * @returns {import('react').JSX.Element}
+ */
+export function CalendarInput({
+  className,
+  dateFormat = 'MM/dd/yyyy',
```

Comment:
This is the default date format if you don't provide one.

Have we tested using a different format?

Reply:
add dateFormat options to the sandbox

Reply:
waiting on UDS-1448